controllers/ingressUpdater.go

```diff
@@ -80,10 +112,47 @@ func (i *ingressUpdater) UpdateIngressOfService(ctx context.Context, service *co
 	return nil
 }
 
-// createCesServiceIngress creates a new ingress resource based on the given ces service.
-func (i *ingressUpdater) createCesServiceIngress(ctx context.Context, cesService CesService, service *corev1.Service, isMaintenanceMode bool) error {
+// updateServiceIngressObject upserts a new ingress resource based on the given ces service.
+func (i *ingressUpdater) updateServiceIngressObject(ctx context.Context, cesService CesService, service *corev1.Service, isMaintenanceMode bool) error {
+	isReady, err := i.deploymentReadyChecker.IsReady(ctx, service.Name)
+	if err != nil {
+		return err
+	}
+
+	err = i.updateIngressObject(ctx, cesService, service, isMaintenanceMode)
+	if err != nil {
+		return err
+	}
+
+	if !isReady {
+		go func() {
+			waitOptions := dogustart.WaitOptions{
+				Timeout:  waitForDeploymentTimeout,
+				TickRate: waitForDeploymentTickRate,
+			}
+			err := i.deploymentReadyReactor.WaitForReady(ctx, service.GetName(), waitOptions, func(ctx context.Context) {
+				log.FromContext(ctx).Info(fmt.Sprintf("dogu [%s] is ready now -> update ingress object", service.GetName()))
+				err = i.updateIngressObject(ctx, cesService, service, isMaintenanceMode)
+				if err != nil {
+					log.FromContext(ctx).Error(err, fmt.Sprintf("failed to update [%s] ingress object", service.GetName()))
+				}
+			})
+			if err != nil {
+				log.FromContext(ctx).Error(fmt.Errorf("failed to wait for the readiness of the [%s] deployment: %w", service.GetName(), err), "failed to update ingress object")
+			}
+		}()
+	}
+
+	return nil
+}
+
+func (i *ingressUpdater) updateIngressObject(ctx context.Context, cesService CesService, service *corev1.Service, isMaintenanceMode bool) error {
```

Comment:
🚫 **[checkstyle]** <funlen> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>Function 'updateIngressObject' is too long (66 > 60)